graphql/src/main/java/com/linecorp/armeria/server/graphql/DataLoaderRegistryCreationStrategy.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.graphql;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+import org.dataloader.DataLoader;
+import org.dataloader.DataLoaderRegistry;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * {@link DataLoaderRegistry} creation strategy that creates a {@link DataLoader}.
+ */
+@UnstableApi
+@FunctionalInterface
+public interface DataLoaderRegistryCreationStrategy {
+
+    /**
+     * Returns a strategy that creates a fixed {@link DataLoaderRegistry}.
+     * Note: Using a fixed {@link DataLoaderRegistry} is not recommended. Instead, use a
```

Comment:
Because this is not recommended, how about not providing this method?

Reply:
If it's okay to make this PR a break change, I'm all for it. 👍

Reply:
We can make `configureDataLoaderRegistry` and `dataLoaderRegistry` mutually exclusive. 
```java
class GraphqlServiceBuilder {

    public GraphqlServiceBuilder dataLoaderRegistry(Function<? super ServiceRequestContext, ? extends DataLoaderRegistry> dataLoaderResistryFactory) {
        ...
    }

    // deprecate configureDataLoaderRegistry

    public GraphqlService build() {

         if (dataLoaderResistryFactory != null && dataLoaderRegistryConsumers != null) {
             throw new IllegalStateException("Can't set dataLoaderResistryFactory and dataLoaderRegistryConsumers together."
         }
    }
```
What do you think of removing `DataLoaderRegistryCreationStrategy` and adding a new builder API using `Function`?

Reply:
We kept the existing API and added a separate function. They are mutually exclusive.